### DISCUSSION TITLE
Minimum Go version issues

### DISCUSSION THREAD

**Discussion by jakereps** [go-webauthn/webauthn#312] [General]
Hey all, I'm seeing some issues with trying to use this library in my service. The README mentions it supports back to 1.21, but due to some dependency updates that landed relatively recently, I appear to be unable to use it on any other Go version than Go 1.23. Was this an intended forced usage of the latest Go version, or can this be reverted in someway to maintain backwards compatibility with currently supported Go versions (n-1 or n-2)?

**Comment by james-d-elliott** [comment]
Yeah we can revert it. I realized a while back that I was incorrectly bumping the version in go.mod, if you want to test it then I'll be happy to make this change (I don't have a single project that's built without the latest go).

**Reply by jakereps** [reply]
Sure I can definitely give it a try, I was able to work around it for now on my project by grabbing v0.11.1 instead of latest, so no rush or urgency on it, and yeah most people develop at the latest tip so probably a majority non-issue, but I have an environment that's unfortunately constrained to n-1 so am temporarily stuck from upgrading beyond the 11.1. The new go.mod behavior is kind of frustrating, tbh 😅

**Reply by james-d-elliott** [reply]
> yeah most people develop at the latest tip so probably a majority non-issue

Yeah this and being lazy and stretched thin are major reasons why I have not fixed it yet.

Give github.com/go-webauthn/webbauthn@2c0872fd86fcc031c1a33192467fcdfa9504c1c9 a try with `GOTOOLCHAIN=auto`.

**Reply by jakereps** [reply]
Super delayed but appear to be getting a 404 on that one.

**Reply by jakereps** [reply]
Aha, two b's in webauthn.
```
go get github.com/go-webauthn/webauthn@build-go-version
go: github.com/go-webauthn/webauthn@v0.11.3-0.20241010023807-2c0872fd86fc requires
        github.com/go-webauthn/x@v0.1.15: module github.com/go-webauthn/x@v0.1.15 requires go >= 1.23 (running go 1.22.8; GOTOOLCHAIN=local)
```

Looks like it's now getting upset over `/x` also being set to >= 1.23

**Reply by james-d-elliott** [reply]
Ah yea I typo'd it. I'll fix up the x module today the same.

**Reply by james-d-elliott** [reply]
Sorry for the delay. Give `e000739f29100fee5b1d341bf95cd42593dae024` a shot.

github.com/go-webauthn/webauthn@e000739f29100fee5b1d341bf95cd42593dae024
